worker.go

```
@@ -85,19 +86,58 @@
 
 // newWorker creates a new worker with the given worker function and configurations
-// The worker function can be any of WorkerFunc, WorkerErrFunc, or VoidWorkerFunc
+// The worker function can be any of WorkerResultFunc, WorkerErrFunc, or WorkerFunc
 // It initializes the worker with the configured concurrency, cache, and other settings
-func newWorker[T, R any](wf any, configs ...any) *worker[T, R] {
+func newWorker[T any](wf WorkerFunc[T], configs ...any) *worker[T, iJob[T]] {
+	c := loadConfigs(configs...)
+
+	w := &worker[T, iJob[T]]{
+		workerFunc: func(j iJob[T]) {
+			utils.WithSafe("worker", func() {
```

golangci-lint (1.64.8)

Error return value of `utils.WithSafe` is not checked

(errcheck)